optimum/neuron/modeling_base.py

```diff
@@ -0,0 +1,440 @@
+        if is_neuron_available() and dynamic_batch_size is True and "batch_size" in input_shapes:
+            input_shapes["batch_size"] = 1
+        neuron_config = neuron_config_constructor(model.config, dynamic_batch_size=dynamic_batch_size, **input_shapes)
+
+        # Get compilation arguments
+        auto_cast_type = None if auto_cast is None else auto_cast_type
+        compiler_kwargs = {
+            "auto_cast": auto_cast,
+            "auto_cast_type": auto_cast_type,
+            "disable_fast_relayout": disable_fast_relayout,
+        }
+
+        input_names, output_names = export(
+            model=model,
+            config=neuron_config,
+            output=save_dir_path / NEURON_FILE_NAME,
+            **compiler_kwargs,
+        )
+
+        store_compilation_config(config, input_shapes, compiler_kwargs, input_names, output_names, dynamic_batch_size)
+
+        config.save_pretrained(save_dir_path)
+        maybe_save_preprocessors(model_id, save_dir_path, src_subfolder=subfolder)
+
+        return cls._from_pretrained(save_dir_path, config, model_save_dir=save_dir, neuron_config=neuron_config)
+
+    def forward(self, *args, **kwargs):
+        raise NotImplementedError
+
+    @classmethod
+    def _auto_model_to_task(cls, auto_model_class):
+        """
+        Get the task corresponding to a class (for example AutoModelForXXX in transformers).
+        """
+        return TasksManager.infer_task_from_model(auto_model_class)
+
+    def _attributes_init(
+        self,
+        model_save_dir: Optional[Union[str, Path, TemporaryDirectory]] = None,
+        preprocessors: Optional[List] = None,
+        **kwargs,
+    ):
+        """
+        Initializes attributes.
+        """
+        self._path_tempdirectory_instance = None
+        if isinstance(model_save_dir, TemporaryDirectory):
+            self._path_tempdirectory_instance = model_save_dir
+            self.model_save_dir = Path(model_save_dir.name)
+        elif isinstance(model_save_dir, str):
+            self.model_save_dir = Path(model_save_dir)
+        else:
+            self.model_save_dir = model_save_dir
+
+        self.preprocessors = preprocessors if preprocessors is not None else []
+
+        self.input_names = getattr(self.config, "input_names", [])
+        self.output_names = getattr(self.config, "output_names", [])
+
+        # Registers the NeuronModelForXXX classes into the transformers AutoModel classes to avoid warnings when creating
+        # a pipeline https://github.com/huggingface/transformers/blob/3d3204c025b6b5de013e07dd364208e28b4d9589/src/transformers/pipelines/base.py#L940
+        AutoConfig.register(self.model_type, AutoConfig)
+        if hasattr(self.auto_model_class, "register"):
+            self.auto_model_class.register(AutoConfig, self.__class__)
+
+    def _neuron_config_init(self, config: "PretrainedConfig"):
+        """
+        Build a Neuron config with an instance of the `PretrainedConfig` and the task.
```

Comment:
```suggestion
        Builds a [`NeuronConfig`] with an instance of the `PretrainedConfig` and the task.
```

Reply:
@michaelbenayoun Why the square brackets?